scripts/linux/electron-installer-appimage.sh

```diff
@@ -102,17 +101,14 @@ else
   exit 1
 fi
 
-if [ -z "$TMPDIR" ]; then
-  TMPDIR=$(mktemp -d)
-fi
-mkdir -p "$TMPDIR"
+APPIMAGE_TMPDIR=$(mktemp --dry-run -d -t "$ARGV_APPLICATION_NAME-linux-$ARGV_ARCHITECTURE-XXXX")
```

Comment:
Creating randomly-named temporary directories is just as make-unfriendly as deleting things!

The idea is to have a (possibly hierarchical) set of consistent (and persistent) dependencies, so that if a target has already been fully-built, then running `make` again won't need to rebuild it; but if one of the build-steps went wrong or got interrupted half-way through, running `make` will just continue from where it left off (which obviously it can't do if directories have a different name each time they're created).

IMHO it probably makes sense to move the directory-creation steps into the `Makefile` too.